File: Compliance Quality Assurance and Tests/MottuFlow/src/main/java/com/sprint/MottuFlow/domain/localidade/Localidade.java
package com.sprint.MottuFlow.domain.localidade;

import com.sprint.MottuFlow.domain.camera.Camera;
import com.sprint.MottuFlow.domain.moto.Moto;
import com.sprint.MottuFlow.domain.patio.Patio;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "localidade")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Localidade {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idLocalidade;

    @Column(nullable = false)
    private LocalDateTime dataHora;

    @Column(nullable = false, length = 100)
    private String pontoReferencia;

    @ManyToOne
    @JoinColumn(name = "id_moto", nullable = false)
    private Moto moto;

    @ManyToOne
    @JoinColumn(name = "id_patio", nullable = false)
    private Patio patio;

    @ManyToOne
    @JoinColumn(name = "id_camera", nullable = false)
    private Camera camera;

	public Long getIdLocalidade() {
		return idLocalidade;
	}

	public void setIdLocalidade(Long idLocalidade) {
		this.idLocalidade = idLocalidade;
	}

	public LocalDateTime getDataHora() {
		return dataHora;
	}

	public void setDataHora(LocalDateTime dataHora) {
		this.dataHora = dataHora;
	}

	public Moto getMoto() {
		return moto;
	}

	public void setMoto(Moto moto) {
		this.moto = moto;
	}

	public Patio getPatio() {
		return patio;
	}

	public void setPatio(Patio patio) {
		this.patio = patio;
	}

	public String getPontoReferencia() {
		return pontoReferencia;
	}

	public void setPontoReferencia(String pontoReferencia) {
		this.pontoReferencia = pontoReferencia;
	}

	public Camera getCamera() {
		return camera;
	}

	public void setCamera(Camera camera) {
		this.camera = camera;
	}
}

File: Advanced Business Development With dotNET/Models/Localidade.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MottuFlowApi.Models
{
    [Table("localidade")]
    public class Localidade
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Column("id_localidade")]
        public int IdLocalidade { get; set; }

        [Required]
        [Column("data_hora")]
        public DateTime DataHora { get; set; } = DateTime.Now;

        [Required]
        [MaxLength(100)]
        [Column("ponto_referencia")]
        public string PontoReferencia { get; set; } = string.Empty;

        [Required]
        [Column("id_moto")]
        public int IdMoto { get; set; }

        [Required]
        [Column("id_patio")]
        public int IdPatio { get; set; }

        [Required]
        [Column("id_camera")]
        public int IdCamera { get; set; }

        // Navigation
        public Moto? Moto { get; set; }
        public Patio? Patio { get; set; }
        public Camera? Camera { get; set; }
    }
}
